aptos-move/mvhashmap/src/versioned_data.rs

```diff
@@ -259,21 +264,41 @@ impl<K: Hash + Clone + Debug + Eq, V: TransactionWrite> VersionedData<K, V> {
         let mut v = self.values.entry(key).or_default();
         let bytes_len = data.bytes_len();
         // For base value, incarnation is irrelevant, set to 0.
-        let prev_entry = v.versioned_map.insert(
-            ShiftedTxnIndex::zero(),
-            CachePadded::new(Entry::new_write_from(0, data, maybe_layout)),
-        );
 
-        assert!(prev_entry.map_or(true, |entry| -> bool {
-            if let EntryCell::Write(i, v, _) = &entry.cell {
-                // base value may have already been provided due to a concurrency race,
-                // but it has to be the same as being set.
-                // Assert the length of bytes for efficiency (instead of full equality)
-                *i == 0 && v.bytes_len() == bytes_len
-            } else {
-                true
-            }
-        }));
+        use btree_map::Entry::*;
+        match v.versioned_map.entry(ShiftedTxnIndex::zero()) {
```

Comment:
Why match over old code? Seems like layout is the only thing that got added?

Reply:
we cannot do insert - that overwrites existing value. we need to do insert only if Vacant.

https://aptos-org.slack.com/archives/C057VPCD7QX/p1697520278313169